lib/Lib.IntTypes.fst

```diff
@@ -322,7 +495,18 @@ let lognot #t #l a =
   | U32  -> UInt32.lognot a
   | U64  -> UInt64.lognot a
   | U128 -> UInt128.lognot a
-
+  | S8 -> Int8.lognot a
+  | S16 -> Int16.lognot a
+  | S32 -> Int32.lognot a
+  | S64 -> Int64.lognot a
+
+let shift_right_negative (#n:pos) (a:Int.int_t n{a < 0}) (s:nat) : Tot (Int.int_t n) =
```

Comment:
This looks like you're manually implementing the arithmetic right shift, unless there's some magic here I'm missing that uses this just for spec and this ends up being extracted as native operators. @s-zanella added support in `FStar.Int` and KreMLin for an arithmetic right shift that extracts to the `>>` operator on compilers that will emit a native arithmetic right shift instruction (`sar` on Intel). How does this extract?

Reply:
Why is this still here?